### PR TITLE
Routing: Update components using props.match to use hooks

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Update the components using `props.match` to use the `useParams` hook instead.

**Why do we need this feature?**
React router upgrade. 

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/93790

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/containers/PublicDashboardPage.test.tsx [hunk 1]
```diff
@@ -55,13 +55,17 @@ jest.mock('app/types', () => ({
...
+jest.mock('react-router-dom-v5-compat', () => ({
+  ...jest.requireActual('react-router-dom-v5-compat'),
+  useParams: jest.fn().mockReturnValue({ accessToken: 'an-access-token' }),
+}));
+
 const setup = (propOverrides?: Partial<Props>, initialState?: Partial<appTypes.StoreState>) => {
   const context = getGrafanaContextMock();
   const store = configureStore(initialState);
-
+  // (useParams as jest.Mock).mockReturnValue({ accessToken: 'an-access-token' });
```
Comment (ashharrison90) [inline public/app/features/dashboard/containers/PublicDashboardPage.test.tsx:66]: comment can be removed?

File: public/app/features/serviceaccounts/ServiceAccountPage.tsx [hunk 2]
```diff
@@ -67,8 +66,9 @@ export const ServiceAccountPageUnconnected = ({
   const [isTokenModalOpen, setIsTokenModalOpen] = useState(false);
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
   const [isDisableModalOpen, setIsDisableModalOpen] = useState(false);
+  const { id = '0' } = useParams();
```
Comment (ashharrison90) [inline public/app/features/serviceaccounts/ServiceAccountPage.tsx:69]: why do we default to `'0'` here? 🤔

**Review by ashharrison90 — APPROVED** [review]
couple of questions, otherwise lgtm 👍

File: public/app/features/serviceaccounts/ServiceAccountPage.tsx (hunk 2 above)
Comment (Clarity-89) [reply public/app/features/serviceaccounts/ServiceAccountPage.tsx:69]: Probably forgot to remove after testing. But tbh it doesn't really matter what the default is, if the param is not there, this component is not going to be rendered.

**Comment by tomratcliffe** [comment]
Can you hold on merging please? Just looking at alerting changes + test approach

**Review by tomratcliffe — APPROVED** [review]
As per Slack, I pushed a change to remove the mocking for the router in the Alerting test with a bit of rework - LGTM! Thanks

File: public/app/features/alerting/unified/RuleEditorExisting.test.tsx [hunk 3]
```diff
@@ -46,7 +46,7 @@ setupMswServer();
 function renderRuleEditor(identifier: string) {
   return render(
     <Routes>
-      <Route path="/alerting/:id/edit" Component={RuleEditor} />
+      <Route path="/alerting/:id/edit" element={<RuleEditor />} />
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/RuleEditorExisting.test.tsx:49]: :+1: Thanks

**Pull request merged**